tests/unit/preprocessor/_time/test_time.py

```diff
@@ -746,6 +746,43 @@ def test_regrid_time_hour(self):
         self.assertArrayEqual(diff_cube.data, expected)
 
 
+class TestTimeseriesFilter(tests.Test):
+    """Tests for regrid_time with hourly frequency."""
+
+    def setUp(self):
+        """Prepare tests."""
+        self.cube = _create_sample_cube()
+
+    def test_timeseries_filter_simple(self):
+        """Test timeseries_filter func."""
+        filtered_cube = timeseries_filter(self.cube, 7, 14)
+        expected_data = np.array(
+            [2.44824568, 3.0603071, 3.67236852, 4.28442994, 4.89649137,
+             5.50855279, 6.12061421, 6.73267563, 7.34473705, 7.95679847,
+             8.56885989, 9.18092131, 9.79298273, 10.40504415, 11.01710557,
+             11.62916699, 12.24122841, 12.85328983]
+        )
+        assert_array_almost_equal(filtered_cube.data, expected_data)
+        assert len(filtered_cube.coord('time').points) == 18
+
```

Comment:
maybe add test with explicit selection of the filter:

``` python

filtered_cube = timeseries_filter(self.cube, 7, 14, filter_type="lowpass")
assert_array_almost_equal(filtered_cube.data, expected_data)
assert len(filtered_cube.coord('time').points) == 18

```

Reply:
good call, we should test the other (future) filter types too; done, cheers :beer: